brushfire-scalding/src/main/scala/com/stripe/brushfire/scalding/Trainer.scala

```diff
@@ -89,7 +90,7 @@ case class Trainer[K: Ordering, V, T: Monoid](
             case (treeIndex, map) => {
               val newTree =
                 treeMap(treeIndex)
-                  .updateByLeafIndex { index => map.get(index).map { t => LeafNode(index, t) } }
+                  .updateByLeafIndex { index => map.get(index).map { t => LeafNode(index, t, Monoid.zero[A]) } }
```

Comment:
Yeah I don't think it's in this PR but definitely the idea is to use metadata to inform splits. (Which probably means adapting `Stopper`, I think).